src/java.xml.crypto/share/classes/com/sun/org/apache/xml/internal/security/algorithms/implementations/SignatureBaseRSA.java

```diff
@@ -584,9 +585,10 @@ public String engineGetURI() {
         PSSParameterSpec pssParameterSpec;
 
         public enum DigestAlgorithm {
-            SHA256("SHA-256", "http://www.w3.org/2001/04/xmlenc#sha256", 32),
-            SHA384("SHA-384", "http://www.w3.org/2001/04/xmldsig-more#sha384", 48),
-            SHA512("SHA-512", "http://www.w3.org/2001/04/xmlenc#sha512", 64);
+//            SHA1("SHA-1", DigestMethod.SHA1, 20),
```

Comment:
Do we want to support "SHA-1"? It's considered weak and not the default but the RFC seems to have not disabled it at all.

Reply:
I'm ok with not supporting SHA-1, although adding it would not be a security issue. It is blocked by default now, but it can be re-enabled, and SHA-1 in general is still available in the JDK.

I'm fine with adding support for SHA-224 and SHA-3 as part of this issue. You can add support for all the algorithms that we have the underlying crypto support for.

Reply:
Not sure if I got it, are you OK with adding SHA-1? It must be listed here to be supported.

Reply:
Let's not add it.